sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py

```diff
@@ -193,8 +191,6 @@ def _wrap_eventdata(
                     raise ValueError(
                         "The partition_key does not match the one of the EventDataBatch"
                     )
-                for message in event_data.message._body_gen:  # pylint: disable=protected-access
-                    add_link_to_send(message, span)
```

Comment:
need `trace_message` for message in the _body_gen?
otherwise it looks like we're not handling OT in this case.